src/detect-engine-analyzer.c

```diff
@@ -861,6 +862,38 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_FLOWBITS: {
+                const DetectFlowbitsData *cd = (const DetectFlowbitsData *)smd->ctx;
+
+                jb_open_object(js, "flowbits");
+                jb_set_uint(js, "idx", cd->idx);
+                jb_set_uint(js, "or_list_size", cd->or_list_size);
+                jb_set_uint(js, "or_list", cd->or_list);
+                if (cd->or_list_size > 0)
+                    jb_set_string(js, "name", cd->or_list[cd->idx]);
```

Comment:
Following the above, I spent some time to understand the code.

Aspects to consider:
- So, cd->idx starts at 1
- cd->or_list_size will only be greater than 0 for rules where we see the flowint or logic (cf https://github.com/OISF/suricata-verify/blob/master/tests/detect-flowbits/test.rules#L3). But the idx is global for flowbits. This means that we could have a list_size 2, and yet the idx for the variables accounted in the or_list be 2 and 3.
- Flowbits will only save a variable name if it sees a rule with `flowbits:set,` with the said variable name. Otherwise, that variable won't be registered, and therefore can't be retrieved with `VarNameStoreSetupLookup`.

So, in the cases when `cd->or_list_size > 0` we will have to that those into account, and then we can open an array to log the variable names. I think we can use the key `ored_variables` here. See https://github.com/OISF/suricata/blob/master/src/detect-engine-analyzer.c#L797 for an example of how to open an array to log a list of values.

This info will be important to be able to properly proceed with this task, and also when improving the accompanying SV tests for them.

Reply:
Am I right to assume that if the `cd->or_list_size > 0` and `cd->cmd == DETECT_FLOWBITS_CMD_SET` then a for loop storing the variable names in an array should begin. I'm not sure about the second condition, should I keep it?

Reply:
No, that's not how this works. `cd->or_list_size` will only be greater than zero for rules like the ones you see in lines 3 and 4 here https://github.com/OISF/suricata-verify/blob/master/tests/detect-flowbits/test.rules#L3-L4

So you should not see `size > 0 ` when `DETECT_FLOWBITS_CMD_SET`. The `SET` command will have only one variable name associated with it. Does this answer your question?

Reply:
okay, so should I create an `ored_variables` array and store names in using a for loop it if `cd->or_list_size > 0`? But I'm not sure on how to access the name if `cd->or_list_size == 0`. Should it be stored in a separate `name` variable?:
```
const char *varname = VarNameStoreSetupLookup(cd->idx, VAR_TYPE_FLOW_BIT);
jb_set_string(js, "name", varname);
```

Reply:
I think this can be simpler.
- No need to create an array, as you could call `VarNameStoreSetupLookup` from the `jb_set_string` or `jb_append_string` functions directly.

- When `or_list_size == 0`, then you know the rule only has one flowbits variable, and therefore you can pass `cd->idx` directly to `VarNameStoreSetupLookup` as id.

Reply:
okay, and these would both go in the `name` variable or should I keep them separate in `ored_variables` and `name`?

Reply:
understood. Thank you so much!